test/functional/Java8andUp/src/org/openj9/test/utf8ClassPkgName/LoadClassWithUTF8PkgNameTest.java

```diff
@@ -0,0 +1,146 @@
+
+package org.openj9.test.utf8ClassPkgName;
+
+import java.util.regex.Pattern;
+import java.util.jar.Manifest;
+import java.util.jar.Attributes;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.File;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+import org.objectweb.asm.*;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.RETURN;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.annotations.BeforeTest;
+
+
+/**
+ * Verify whether VM successfully loads a class with a Chinese UTF8 package name from a jar file
+ * Note: The specified jar file to to be loaded in this test is automatically generated so as to
+ * avoid any issue with platform dependencies.
+ * 
+ */
+@Test(groups = { "level.sanity" })
+public class LoadClassWithUTF8PkgNameTest {
+	
+	private static byte[] classBytes;
+	final static int classVersionJava8 = 52;
+	final static String loadedPackagePath = "org/openj9/resources/utf8pkgname/loaded_Utf8Class_\u6b63\u5728\u52a0\u8f7d\u7c7b\u5305\u540d\u957f\u5ea6\u6570\u68c0\u67e5/";
+	final static String loadedClassName = "Utf8ClassPackageNameTest";
+	final static String loadedClassFileWithPackagePath = loadedPackagePath + loadedClassName + ".class";
+	final static String loadedClassNameWithDotPath = loadedPackagePath.replace('/', '.') + loadedClassName;
+	final static String jarFileNameWithPath = System.getProperty("user.dir") + "/" + "Utf8PkgName.jar";
+	
+	public static byte[] generateClassBytes(String loadedPackagePath, String loadedClassName, int classVersion) throws Exception {
+
+		ClassWriter cw = new ClassWriter(0);
+		FieldVisitor fv;
+		MethodVisitor mv;
+		AnnotationVisitor av0;
+
+		cw.visit(classVersion, ACC_PUBLIC + ACC_SUPER, loadedPackagePath + loadedClassName, null, "java/lang/Object", null);
+		cw.visitSource(loadedClassName + ".java", null);
+
+		{
+			mv = cw.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
+			mv.visitCode();
+			Label l0 = new Label();
+			mv.visitLabel(l0);
+			mv.visitLineNumber(29, l0);
+			mv.visitVarInsn(ALOAD, 0);
+			mv.visitMethodInsn(INVOKESPECIAL, "java/lang/Object", "<init>", "()V");
+			mv.visitInsn(RETURN);
+			Label l1 = new Label();
+			mv.visitLabel(l1);
+			mv.visitLocalVariable("this", "L" + loadedPackagePath + loadedClassName + ";", null, l0, l1, 0);
+			mv.visitMaxs(1, 1);
+			mv.visitEnd();
+		}
+		cw.visitEnd();
+
+		return cw.toByteArray();
+	}
+	
+	@BeforeTest
+	public void beforeTest() throws Exception {
+		String ver = System.getProperty("java.version");
+		boolean isJava8ClassFile = ((ver.length()>=3) && ver.substring(0, 2).equalsIgnoreCase("1.8")) ? true : false;
```

Comment:
There is a helper that does this, can you use org.openj9.test.util.VersionCheck.getMajor() instead?

Reply:
Replaced with `VersionCheck.major()` and verified locally with internal JDK8, OpenJDK Java 11 and 14.